File: 백준/Gold/1644. 소수의 연속합/소수의 연속합.js
const fs = require('fs');

const read = fs.readFileSync('dev/stdin').toString().trim();

const n = Number(read);

const get_primes = () => {
    const li = new Array(n + 1).fill(true);
    li[0] = false; li[1] = false;
    for (let i = 2; i < n + 1; i ++ ){
        if (li[i]) {
            for (let j = i * 2; j < n + 1; j += i) {
                li[j] = false;
            }
        }
    }
    
    return Array.from(li).map((_, i) => i).filter(i => li[i] && i >= 2);
}

const main = () => {
    const primes = get_primes()
    
    let left = 0
    let right = 0;
    let sumv = 0;
    let cnt = 0;
    while (right <= primes.length) {
        if (sumv < n){
            if(right >= primes.length) break;
            sumv += primes[right++];
            continue;
        }
        
        if (sumv == n){
            cnt += 1;
        }
        sumv -= primes[left++];
    }
    
    return cnt;
}

console.log(main())

File: 백준/Gold/1644. 소수의 연속합/소수의 연속합.py
n = int(input())

def get_prime(n):
    li = [True] * (n + 1)
    for i in range(2, n):
        if li[i]:
            for j in range(i ** 2, n + 1, i):
                li[j] = False
                
    return [i for i in range(2, n + 1) if li[i]]


def main():
    primes = get_prime(n)
    left = 0
    right = 0
    sumv = 0
    
    cnt  = 0
    
    while right <= len(primes):
        if sumv < n:
            if right >= len(primes):
                break
            sumv += primes[right]
            right += 1
            continue
            
        if sumv == n:
            cnt += 1
        sumv -= primes[left]
        left += 1
    
    return cnt

print(main())
        